lib/shimmer.dart

```diff
@@ -125,29 +125,31 @@ class Shimmer extends StatefulWidget {
 }
 
 class _ShimmerState extends State<Shimmer> with SingleTickerProviderStateMixin {
-  late AnimationController _controller;
+  late final AnimationController _controller;
   int _count = 0;
 
   @override
   void initState() {
     super.initState();
     _controller = AnimationController(vsync: this, duration: widget.period)
-      ..addStatusListener((AnimationStatus status) {
-        if (status != AnimationStatus.completed) {
-          return;
-        }
-        _count++;
-        if (widget.loop <= 0) {
-          _controller.repeat();
-        } else if (_count < widget.loop) {
-          _controller.forward(from: 0.0);
-        }
-      });
+      ..addStatusListener(controllerStateListener);
     if (widget.enabled) {
       _controller.forward();
     }
   }
 
+  void controllerStateListener(AnimationStatus status) {
```

Comment:
This method should be private and the method name should start with underscore (as convention)

Reply:
is Done! Thx for hint